source/slang/slang-ir.cpp

```diff
@@ -2524,6 +2532,14 @@ namespace Slang
         }
     }
 
+    IRInst* IRBuilder::emitGetValueFromExistentialBox(IRType* type, IRInst* existentialBox)
+    {
+        auto inst =
+            createInst<IRInst>(this, kIROp_GetValueFromExistentialBox, type, 1, &existentialBox);
```

Comment:
I guess technically the `getValueFromExistentialBox` instruction is effectively just a `load` because the `ExistentialBox<T>` type is a kind of pointer type.

I suspect that having a distinct op here makes sense, but I wonder if maybe I need to remove the pointer-ness from the box type eventually.

I think I made `ExistentialBox<T>` act like a `Ptr<T>` because at the time I thought that a value of existential type could just be a single pointer, and that would mean that a `Ptr<Derived>` would have the same layout as an `IBase`.

That is very much *not* the case today, both because an `IBase` is now a tuple, and also because the bits of the `Derived` value might not be stored indirectly.

As such, the pointer-like-ness of `ExistentialBox` is debatable at best. The main argument in favor of it is just for the "legacy" layout model where we might need to shuffle around data to make up for the lack of pointers.

I expect that in a future evolution of this logic, we might need to make a clean separation between `ExistentialBox<Bar, IFoo>`, which is simply indicating a type that is layout-compatible with `IFoo`, but that is known to contain a `Bar` in its any-value part, and a separate type named something like `PseudoPtr<Bar>` which represents the idea of a *zero-size* field that logically acts as a pointer to a value of type `Bar`.